src/client/datascience/jupyter/jupyterVariables.ts

```diff
@@ -65,9 +65,15 @@ export class JupyterVariables implements IJupyterVariables {
     public async getDataFrameInfo(
         targetVariable: IJupyterVariable,
         notebook?: INotebook,
-        sliceExpression?: string
+        sliceExpression?: string,
+        shouldUpdateCachedInfoCachedInfo?: boolean
```

Comment:
Suggestion: Add new method for clearing the cache instead. Makes this method have fewer responsibilities.

Reply:
Actually that doesn't work as well. What would you clear? If it's on this item it forces a refresh of just this variable.

So ignore the suggestion.

Reply:
Would it be better if I called getFullVariable in the jupyterVariableDataProvider, and then passed that result to getDataFrameInfo in jupyterVariables.ts, since kernelVariables and debuggerVariables do not cache anything? The flag name feels like a misnomer at a minimum.